vowpalwabbit/object_pool.h

```diff
@@ -121,30 +121,23 @@ struct no_lock_object_pool
   std::queue<T*> m_pool;
 };
 
-template <typename T, typename TAllocator, typename TDeleter>
-struct value_object_pool
+template <typename T>
+struct moved_object_pool
 {
-  value_object_pool() = default;
+  moved_object_pool() = default;
+
+  void reclaim_object(T&& obj) { m_pool.push(std::move(obj)); }
 
-  ~value_object_pool()
+  void acquire_object(T& dest)
```

Comment:
I'm not sure I'm a fan of this syntax. It would probably be cleaner to use `unique_ptr` with a custom deleter that returns the objects to the pool, but thats a much more involved change

Reply:
It is a much more involved change thats for sure. I would like to avoid this for now